src/xenia/base/filesystem_posix.cc

```diff
@@ -156,7 +153,7 @@ bool CreateFile(const std::filesystem::path& path) {
 
 bool DeleteFile(const std::filesystem::path& path) {
   // TODO: proper implementation.
-  return (path.c_str()) == 0 ? true : false;
+  return path.c_str() == nullptr;
```

Comment:
[`unlink()`](https://linux.die.net/man/2/unlink) ?

Reply:
[remove()](https://en.cppreference.com/w/cpp/filesystem/remove)?

Reply:
Ok, Std is (almost) always better than sys functions

Reply:
Although I am not sure if we should allow folders getting deleted which remove does. (Contrary to unlink). 
Check if its a folder and return false? (And some log message)

Reply:
I'm wondering if there's any point to having this function anymore.
There are [one place](https://github.com/xenia-project/xenia/blob/master/src/xenia/vfs/devices/host_path_entry.cc#L105) where `DeleteFile` is called and it is specifically expecting `path` to  not be a directory.

It could easily be replaced with
```c++
// Delete File.
return !std::filesystem::is_directory(full_path) && std::filesystem::remove(full_path);
```

Unless there is a reason not to do this, I would like to remove `IsFolder`, `PathExists`, `CreateFolder`, `DeleteFolder` and `DeleteFile` with `std::filesystem` functions in the place where they are called.

I realize that there might be a reason not to do this. Slippery slope argument: the same thing could be done with file handle functions and even with `std::threads`. So I am experimenting and waiting for feedback.

Reply:
About the slippery slope:
It depends. For example I decided against using `std::chrono::high_resolution_clock` for the rework of the clock code because I wanted full control over the arithmetics. Using `std` lib could have resulted in two rescales. The risk of that going ooopsie in some games was too high for my taste.
There could be similar arguments with `std::threads` for example. Reading std documentation may not be enough, for chrono stuff I even peeked into the MS implementation of it...

Reply:
PR for those functions is up
https://github.com/xenia-project/xenia/pull/1589